sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -78,6 +78,27 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     }
   }
 
+  private def toSQL(node: LogicalPlan, topNode: Boolean): String = {
```

Comment:
If my understanding is not wrong, you want to support SQL generation from a partial analyzed plan. If so, do you have any use case?

So far, our SQL generation assumes the plan is complete and generated by the SQL parser. If not, I am afraid, it could generate a wrong SQL statement.